Mueez Afzal-weatherman.py

```diff
@@ -0,0 +1,304 @@
+          else:
+              for i in range(self.highest):
+                  print ("+",end="")
+          print(" ",self.highest,"C")
+          print("Coldest temperature:",end="")
+          if self.lowest==None:
+              print("NO Record")
+          elif self.lowest<=0:
+              for i in range(self.lowest,0):
+                 print ("-",end="")
+          else:
+              for i in range(self.lowest):
+                 print ("-",end="")
+          print(" ",self.lowest,"C")
+
+def file_data(day_list,month_list,year_list):
+    day_count,month_count,year_count=0,0,0
+    hot_list=[]
+    cold_list=[]
+    humidity_list=[]
+
+    for filename in os.listdir("weatherfiles"):
+        if filename.endswith(".txt"):
+            year_name=filename.split("_")
+
+            if year_count==0 or year_list[year_count-1].get_name() != year_name[2]:
+
+                year_list.append(year_count)
+                year_list[year_count]=year()
+                year_list[year_count].set_name(year_name[2])
+                year_count+=1 
+
+            filename=("weatherfiles/"+filename)
+            file=open(filename,"r")
+            line=file.readline()
+
+            month_list.append(month_count)
+            month_list[month_count]=month()
+            month_list[month_count].set_year(year_name[2])
+                 
+            for j in range(0,32):
+                line=file.readline()
+                line=line.split(',')
+
+                if line[0] !="":
+
+                    date=line[0].split("-")
+                    month_name=months.get(date[1])
+
+                    day_list.append(day_count)
+                    day_list[day_count]=day()
+                    day_list[day_count].set_data(year_name[2],month_name,date[2])
+            
+                    if line[1] !="":
+                        day_list[day_count].set_hot(int(line[1]))
+                        hot_list.append(int(line[1]))
+
+                        if year_list[year_count-1].check_hot(int(line[1])):
+                           year_list[year_count-1].set_highest(month_name,date[2],int(line[1]))
+
+                    if line[3] !="":
+                        day_list[day_count].set_cold(int(line[3]))
+                        cold_list.append(int(line[3]))
+
+                        if year_list[year_count-1].check_cold(int(line[3])):
+                           year_list[year_count-1].set_lowest(month_name,date[2],int(line[3]))
+
+                    if line[8] !="":
+                        humidity_list.append(int(line[8]))
+
+                    if line[7] !="":
+                       if year_list[year_count-1].check_humidity(int(line[7])):
+                          year_list[year_count-1].set_humidity(month_name,date[2],int(line[7]))
+
+                    day_count+=1
+
+            avg_hot=sum(hot_list)/len(hot_list)
+            avg_cold=sum(cold_list)/len(cold_list)
+            avg_humidity=sum(humidity_list)/len(humidity_list)

```

Comment:
Please follow the single responsibility principle. Since the responsibility of this method should only read data and store it in the field of this class as a list, we shouldn't calculate results here.